thefuck/types.py

```diff
@@ -255,4 +255,4 @@ def run(self, old_cmd):
         logs.debug(u'PYTHONIOENCODING: {}'.format(
             os.environ.get('PYTHONIOENCODING', '!!not-set!!')))
 
-        print(self._get_script())
+        sys.stdout.write(self._get_script())
```

Comment:
does windows not obey the end param to print?  i.e. if we set `print(self._get_script(), end='\n')` would it work?  I dont have a windows box to test this on...

Reply:
Thanks for reviewing, @davidhart82.

It does. Thing is, that usage of `print` is not Python 2 compatible.

Now that Python 2 is sunset, It'd be good to take the next release as an opportunity to make it the last one compatible with Python 2. Hence the use of `sys.stdout.write` instead of `print`.

Reply:
Not even with `from __future__ import print_function`?

Reply:
Of course, it does. But that would be more intrusive, I think. One more thing that wouldn't make sense moving forward, supporting only Python 3.

I'd rather go with `sys.stdout.write`. Works the very same with 2 & 3. Works moving forward. No additional import.

Any particular reason why you believe `print` should be maintained?

Reply:
Ha, no, sorry, didn't mean to sound confrontational, just wanted to understand the rationale.

Reply:
Neither did I, sorry. I just started thinking that I might be missing something.

At some point, I wondered if a call to `sys.stdout.flush` would be necessary right after that. But then, the process exits right after that, so it would be kind of redundant.

While writing this, I thought about looking into integration tests and see if there's something we should assert that.

Reply:
> At some point, I wondered if a call to sys.stdout.flush would be necessary right after that. But then, the process exits right after that, so it would be kind of redundant.

What are your thought, @davidhart82?

Reply:
> At some point, I wondered if a call to `sys.stdout.flush` would be necessary right after that. But then, the process exits right after that, so it would be kind of redundant.

I think if it wasn't necessary for `print` statement, it wouldn't be necessary now either.

Reply:
@scorphus and @davidhart82 
Should I resolve this conversation? This has been unattended since a month.

Reply:
I guess so.

Hey, @davidhart82, we still haven't read from you! 🙂

Reply:
Yes, sorry. Looks fine. Print doesn't cause a flush by default either, so no change in behaviour should be expected.

Reply:
Thanks for the review!